go-controller/pkg/ovn/dnsnameresolver.go

```diff
@@ -0,0 +1,104 @@
+package ovn
```

Comment:
I think this file should go into `dns-name-resolver` folder and use its own fields instead of DefaultNetworkController

Reply:
This file is related to the level-driven controller for DNSNameResolver resources. The `reconcileDNSNameResolver` uses `DefaultNetworkController`'s fields. If we move the file into `dns-name-resolver` package it'll create import loop.

Reply:
I am not sure I understand, which import loop you mean. I am suggesting to move all `DefaultNetworkController` dns-related fields and `reconcileDNSNameResolver`, and `initControllers` inside the implementation of `DNSNameResolver`. So that you only need to call `DNSNameResolver.Run` and it will start controllers and reconcilers itself.
About the interface, I would expect it to be very similar to the old dns resolver interface, and all the internal structs update logic together with waiting for the address set to be created would be hidden inside the new dns resolver implementation.
Maybe that is impossible and I am missing why, please let me know.

Reply:
I misunderstood your last comment. I have made the required changes and moved the code to `dns-name-resolver` package.